xarray/tests/test_plot.py

```diff
@@ -89,6 +89,20 @@ class TestPlot(PlotTestCase):
     def setUp(self):
         self.darray = DataArray(easy_array((2, 3, 4)))
 
+    def test_label_from_attrs(self):
+        da = self.darray
+        da.name = 'a'
+        da.attrs['units'] = 'a_units'
+        da.attrs['long_name'] = 'a_long_name'
+        da.attrs['standard_name'] = 'a_standard_name'
+        assert 'a_long_name [a_units]' == label_from_attrs(da)
+        da.attrs.pop('long_name')
+        assert 'a_standard_name [a_units]' == label_from_attrs(da)
+        da.attrs.pop('standard_name')
+        assert 'a [a_units]' == label_from_attrs(da)
+        da.attrs.pop('units')
+        assert 'a' == label_from_attrs(da)
```

Comment:
perfect test

Reply:
Good test, sure, but it doesn't cover what happens if the units are present without various combinations of name metadata.

I'd also find this easier to follow using [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/reference.html#pytest-mark-parametrize) to separate each combination.